src/domain/event/__tests__/CreateEventPage.test.tsx

```diff
@@ -481,6 +473,174 @@ const fillForm = async (eventFormData: Partial<CreateEventFormFields>) => {
+  const transletableFieldLabels = [
+    /^Tapahtuman nimi/,
+    /^Lyhyt kuvaus \(korkeintaan 160 merkkiä\)/,
+    /^Kuvaus/,
+    /^WWW-osoite, josta saa lisätietoja tapahtumasta/,
+    /^Lisätiedot/,
+  ];
+
+  beforeEach(() => {
+    jest.resetModules();
+  });
+
+  it('has Finnish, Swedish and English as language options', async () => {
+    render(<CreateEventPage />, { mocks });
+    const languageSelector = await screen.findByTestId(
+      formLanguageSelectorTestId
+    );
+    within(languageSelector).getByText(/Valitse lomakkeen kieliversiot/i);
+    // Finnish should be selected by default
+    expect(
+      await within(languageSelector).getByLabelText(/suomi/i)
+    ).toBeChecked();
+    // Rest of the langauges should be unchecked by default
+    expect(
+      within(languageSelector).getByLabelText(/ruotsi/i)
+    ).not.toBeChecked();
+    expect(
+      within(languageSelector).getByLabelText(/englanti/i)
+    ).not.toBeChecked();
+  });
+
+  test('transletable fields appears in selected languages', async () => {
+    render(<CreateEventPage />, { mocks });
+    const languageSelector = await screen.findByTestId(
+      formLanguageSelectorTestId
+    );
+    // Select Swedish (with Finnish that is already selected)
+    userEvent.click(within(languageSelector).getByLabelText(/ruotsi/i));
+
+    transletableFieldLabels.forEach((labelText) => {
+      screen.getByLabelText(new RegExp(labelText.source + / \(FI\)/.source));
+      screen.getByLabelText(new RegExp(labelText.source + / \(SV\)/.source));
+    });
+
+    // Unselect Finnish
+    userEvent.click(within(languageSelector).getByLabelText(/suomi/i));
+    transletableFieldLabels.forEach((labelText) => {
+      expect(
+        screen.queryByLabelText(new RegExp(labelText.source + / \(FI\)/.source))
+      ).not.toBeInTheDocument();
+    });
+  });
+
+  /*
+    FIXME: It would be better to spy on method that manipulates values 
+    and check that the values of unselect language are empty.
+    Doing so would make the unit test faster and smaller 
+    and it would test the right spot.
+    It could be achieved by doing something like this:
+    ```
+      const omitUnselectedLanguagesFromValuesSpy = jest.spyOn(
+        Utils,
+        'omitUnselectedLanguagesFromValues'
+      );
+      // ...
+      // Check the spyOn return values https://github.com/facebook/jest/issues/3821
+      expect(omitUnselectedLanguagesFromValuesSpy).toHaveBeenCalled();
+      expect(
+        omitUnselectedLanguagesFromValuesSpy.results[0].name.fi
+      ).toHaveValue();
+      expect(omitUnselectedLanguagesFromValuesSpy.results[0].name.sv).toHaveValue(
+        ''
+      );
+    ```
+  */
+  test('filled fields for unselected languages are cleared when submitting the form', async () => {
+    const pushMock = mockUseHistory();
+
+    const genericSwedishValue = 'SV translation';
+
+    render(<CreateEventPage />, {
+      mocks: mocks,
+    });
+
+    await fillForm({ ...defaultFormData });
```

Comment:
could be simplified, no spread needed: `await fillForm(defaultFormData);` this seems to have been like this (with spread) before your changes.